src/contexts/TodoContext.js

```diff
@@ -0,0 +1,66 @@
+import React, { createContext, useState } from 'react'
+
+const TodoContext = createContext()
+
+const TodoProvider = ({ children }) => {
+  const [todos, setTodos] = useState([])
+  const [todo, setTodo] = useState('')
+  const [statusTodo, setStatusTodo] = useState('all')
+
+  const addTodoKeyDown = (event) => {
+    if (event.keyCode === 13 && event.target.value !== '') {
+      addTodo()
+      setTodo('')
+    }
+  }
+
+  const addTodo = () => {
+    const newTodo = { title: todo, status: 'pending' }
+    setTodos([...todos, newTodo])
+  }
+
+  const addTodoButtonClick = () => {
+    if (todo !== '') {
+      addTodo()
+      setTodo('')
+    }
+  }
+
+  const setTodoChange = (event) => setTodo(event.target.value)
+
+  const deleteIconClick = (index) => {
+    const copyTodos = todos.filter((todo) => todos.indexOf(todo) !== index)
+    setTodos([...copyTodos])
+  }
+
+  const todoStatusChange = (index) => {
```

Comment:
Por convención, el verbo en el nombre de la función va primero (ya que es lo que hace)

```suggestion
  const changeTodoStatus = (index) => {
```